dtfs-central-api/src/services/repositories/utilisation-reports-repo/utilisation-reports-repo.ts

```diff
@@ -17,7 +18,7 @@ export const saveUtilisationReportDetails = async (reportPeriod: ReportPeriod, a
     reportPeriod,
     dateUploaded: new Date(),
     azureFileInfo,
-    status: UTILISATION_REPORT_RECONCILIATION_STATUS.PENDING_RECONCILIATION,
+    status: 'PENDING_RECONCILIATION',
```

Comment:
This appears to have been moved back from constant, any specific reason?

Reply:
Yes - in typescript files, if the type is a `string` literal union, you are only able to set the value to one of the strings specified by the type. As the type is extracted from the constant, we don't need to use the constant explicitly in typescript files as they will always be in sync.

In a typescript first project, you wouldn't need a constant for `string`-enum things like this and would only have the type:
```
type UtilisationReportReconciliationStatus =
  | 'PENDING_RECONCILIATION'
  | 'REPORT_NOT_RECEIVED'; // etc

/**
  * intellisense shows either 'PENDING_RECONCILIATION' or 'REPORT_NOT_RECEIVED' and
  * you get a TS-error if you put anything else
  */
const status: UtilisationReportReconciliationStatus = 
```

I imagine we never would have introduced the `UTILISATION_REPORTS_RECONCILIATION_STATUS` constant if we didn't need to use it in some javascript files, so removing the use of the constant where we can is done with the hope of being able to permanently remove it eventually

Reply:
Happy to approve this, however in my opinion repeating status multiple times is not keeping source code DRY.
One for a wider discussion we should look into following links for references:

* https://github.com/Microsoft/TypeScript/pull/6532
* https://www.typescriptlang.org/docs/handbook/enums.html
* https://github.com/UK-Export-Finance/exip

Reply:
My opinion on this (FWIW) is that this doesn't go against DRY principles at all. You're either repeating
```
UTILISATION_REPORT_RECONCILIATION_STATUS.PENDING_RECONCILIATION
```
or you're repeating
```
'PENDING_RECONCILIATION'
```
The constant is redundant, and doesn't add any value when using TypeScript and a union type.

If the type of `status` was `string`, or we were writing plain JavaScript then the constant _does_ add value because it limits the values to those set by the constant, and makes refactoring easier.

But that's not the case here. The type of `status` is a union type
```
export type UtilisationReportReconciliationStatus =
  | 'REPORT_NOT_RECEIVED'
  | 'PENDING_RECONCILIATION'
  | 'RECONCILIATION_IN_PROGRESS'
  | 'RECONCILIATION_COMPLETED';
```
so with TypeScript you can only set one of the above values. Your IDE provides you with auto-complete for these values, and if you try to set anything else TypeScript will shout at you and the code will fail to compile.

This is similar to the `boolean` type which is a union of `true | false`. In that case there isn't any need to use a constant (e.g. `BOOLEAN.TRUE` or `BOOLEAN.FALSE`) and the same principle can be applied here.